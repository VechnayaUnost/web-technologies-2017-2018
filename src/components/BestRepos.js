import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { fetchBestReposRequest } from '../actions';

import ReposList from './ReposList';

class BestRepos extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            reposType: 'popular'
        };
    }

    componentDidMount() {
        this.handleReposSearch(this.state.reposType);
    }

    changeReposType(e) {
        this.setState({
            reposType: e.target.value
        })
        this.handleReposSearch(e.target.value);
    }

    handleReposSearch(reposType) {
        this.props.fetchBestReposRequest(reposType)
    }

    render() {
        return (
            <div>
                <h1>Best Repositories by: {this.state.reposType}</h1>
                <div className='inputs'>
                    <button value={'popular'} onClick={this.changeReposType.bind(this)} className={'button ' + (this.state.reposType === 'popular' ? 'active' : '')}>Popular</button>
                    <button value={'trending'} onClick={this.changeReposType.bind(this)} className={'button ' + (this.state.reposType === 'trending' ? 'active' : '')}>Trending</button>
                </div>
                <ReposList repositories={this.props.repositories}/>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        repositories: state.bestRepos,
    };
};

const mapDispatchToProps = dispatch => bindActionCreators({
    fetchBestReposRequest,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(BestRepos);