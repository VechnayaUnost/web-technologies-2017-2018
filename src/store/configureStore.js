import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware, { END } from 'redux-saga';

import rootReducer from '../reducers';
import rootSaga from '../sagas';

export default (initialState) => {
    const sagaMiddleware = createSagaMiddleware();
    const store = createStore(rootReducer, initialState, applyMiddleware(sagaMiddleware));

    sagaMiddleware.run(rootSaga);
    store.runSaga = () => sagaMiddleware.run(rootSaga);
    store.close = () => store.dispatch(END);

    return store;
};