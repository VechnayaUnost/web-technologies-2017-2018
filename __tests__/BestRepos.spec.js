import React from 'react';
import BestRepos from '../src/components/BestRepos';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

configure({ adapter: new Adapter() });

describe('BestRepos component test', () => {
    const mockStore = configureStore([thunk]);
    const initialState = {
        bestRepos: [
            {
                name: 'AvatarSkeleton',
                description: 'Creates Avatar skeleton with some preinstalled options',
                stargazers_count: 321,
                watchers_count: 123,
                language: 'C#',
                fork: true,
                archived: false
            },
            {
                name: 'AvatarMountConnection',
                description: 'Avatar mount connection is easier with this tool',
                stargazers_count: 543,
                watchers_count: 345,
                language: 'Java',
                fork: false,
                archived: true
            },
            {
                name: 'SoldiersBarracks',
                description: 'Barrack project for human soldiers',
                stargazers_count: 2,
                watchers_count: 1,
                language: 'Paper&Pen',
                fork: false,
                archived: false
            }
        ]
    }

    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    })

    it('BestRepos component rendered succesfully', () => {
        const  bestRepos = mount(
            <BestRepos store={store} />
        );
        expect(mountToJson(bestRepos)).toMatchSnapshot();
    });

    it('BestRepos component event triggers', () => {
        const props = {
            fetchBestReposRequest: jest.fn()
        }
        const bestRepos = mount(
            <BestRepos store={store} {...props}/>
        );
        bestRepos.setState({text: 'meow'});
        bestRepos.find('button').not('.active').simulate('click');
        bestRepos.instance().props.fetchBestReposRequest(bestRepos.find('button').not('.active').simulate('click'));
        expect(props.fetchBestReposRequest).toHaveBeenCalled();
    });
});