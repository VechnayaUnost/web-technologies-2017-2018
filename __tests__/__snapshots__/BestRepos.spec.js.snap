// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BestRepos component test BestRepos component rendered succesfully 1`] = `
<Connect(BestRepos)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BestRepos
    fetchBestReposRequest={[Function]}
    repositories={
      Array [
        Object {
          "archived": false,
          "description": "Creates Avatar skeleton with some preinstalled options",
          "fork": true,
          "language": "C#",
          "name": "AvatarSkeleton",
          "stargazers_count": 321,
          "watchers_count": 123,
        },
        Object {
          "archived": true,
          "description": "Avatar mount connection is easier with this tool",
          "fork": false,
          "language": "Java",
          "name": "AvatarMountConnection",
          "stargazers_count": 543,
          "watchers_count": 345,
        },
        Object {
          "archived": false,
          "description": "Barrack project for human soldiers",
          "fork": false,
          "language": "Paper&Pen",
          "name": "SoldiersBarracks",
          "stargazers_count": 2,
          "watchers_count": 1,
        },
      ]
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div>
      <h1>
        Best Repositories by: 
        popular
      </h1>
      <div
        className="inputs"
      >
        <button
          className="button active"
          onClick={[Function]}
          value="popular"
        >
          Popular
        </button>
        <button
          className="button "
          onClick={[Function]}
          value="trending"
        >
          Trending
        </button>
      </div>
      <ReposList
        repositories={
          Array [
            Object {
              "archived": false,
              "description": "Creates Avatar skeleton with some preinstalled options",
              "fork": true,
              "language": "C#",
              "name": "AvatarSkeleton",
              "stargazers_count": 321,
              "watchers_count": 123,
            },
            Object {
              "archived": true,
              "description": "Avatar mount connection is easier with this tool",
              "fork": false,
              "language": "Java",
              "name": "AvatarMountConnection",
              "stargazers_count": 543,
              "watchers_count": 345,
            },
            Object {
              "archived": false,
              "description": "Barrack project for human soldiers",
              "fork": false,
              "language": "Paper&Pen",
              "name": "SoldiersBarracks",
              "stargazers_count": 2,
              "watchers_count": 1,
            },
          ]
        }
      >
        <ul>
          <Repo
            repo={
              Object {
                "archived": false,
                "description": "Creates Avatar skeleton with some preinstalled options",
                "fork": true,
                "language": "C#",
                "name": "AvatarSkeleton",
                "stargazers_count": 321,
                "watchers_count": 123,
              }
            }
          >
            <li
              className="repo"
            >
              <div
                className="repo-description"
              >
                <h3>
                  AvatarSkeleton
                </h3>
                <p>
                  Creates Avatar skeleton with some preinstalled options
                </p>
              </div>
              <div
                className="repo-stars-lang"
              >
                <span
                  className="repo-lang"
                >
                  Language: 
                  C#
                </span>
                <span
                  className="repo-stars"
                >
                  Stars: 
                  321
                </span>
              </div>
            </li>
          </Repo>
          <Repo
            repo={
              Object {
                "archived": true,
                "description": "Avatar mount connection is easier with this tool",
                "fork": false,
                "language": "Java",
                "name": "AvatarMountConnection",
                "stargazers_count": 543,
                "watchers_count": 345,
              }
            }
          >
            <li
              className="repo"
            >
              <div
                className="repo-description"
              >
                <h3>
                  AvatarMountConnection
                </h3>
                <p>
                  Avatar mount connection is easier with this tool
                </p>
              </div>
              <div
                className="repo-stars-lang"
              >
                <span
                  className="repo-lang"
                >
                  Language: 
                  Java
                </span>
                <span
                  className="repo-stars"
                >
                  Stars: 
                  543
                </span>
              </div>
            </li>
          </Repo>
          <Repo
            repo={
              Object {
                "archived": false,
                "description": "Barrack project for human soldiers",
                "fork": false,
                "language": "Paper&Pen",
                "name": "SoldiersBarracks",
                "stargazers_count": 2,
                "watchers_count": 1,
              }
            }
          >
            <li
              className="repo"
            >
              <div
                className="repo-description"
              >
                <h3>
                  SoldiersBarracks
                </h3>
                <p>
                  Barrack project for human soldiers
                </p>
              </div>
              <div
                className="repo-stars-lang"
              >
                <span
                  className="repo-lang"
                >
                  Language: 
                  Paper&Pen
                </span>
                <span
                  className="repo-stars"
                >
                  Stars: 
                  2
                </span>
              </div>
            </li>
          </Repo>
        </ul>
      </ReposList>
    </div>
  </BestRepos>
</Connect(BestRepos)>
`;
